Readme.md

```diff
@@ -0,0 +1,181 @@
+
+### Virtual Network for Azure Front Door Premium
+
+![Azure Front Door Premium with Azure Bastion Network Diagram](docs/media/network-diagram-ingress-with-vnet.png)
+
+Because the storage accounts are locked down, the sample also allows you to deploy an Azure Virtual Network with Azure Bastion and a Jumpbox Virtual Machine. Private endpoint connections are made to the storage account in the Virtual Network. This will allow you to interact with the Storage Accounts from the Jumpbox.
+
+Again, all requests will flow through Azure Front Door and Azure Front Door will connect to the Azure Storage Accounts through the Microsoft Backbone network via Private Link.
+
+This deployment illustrates how an administrator could connect to a jumpbox virtual machine via Azure Bastion to perform any (emergency) administrative tasks. The following resources are deployed:
+
+- Azure Virtual Network with 5 subnets:
+  - default
+  - FirewallSubnet
+  - PrivateEndpointsSubnet
+  - JumpboxSubnet
+  - AzureBastionSubnet
+- Jumpbox Virtual Machine
+- Azure Bastion
+- Private endpoints to each storage account
+- Private DNS Zone - configured with A records for the private endpoint URLs to the storage accounts
+
+The following describes the administrative flow:
+
+1. An administrator connects to Azure Bastion that is deployed in the Virtual Network.
+2. Azure Bastion provides SSH connectivity to the jumpbox virtual machine.
+3. The administrator on the jumpbox tries to access the storage account via the Azure CLI. The jumpbox queries DNS for the public Azure Blob Storage Account endpoint: storageaccountname.blob.core.windows.net. Private DNS ultimately resolves to storageaccountname.privatelink.blob.core.windows.net, returning the private IP address of the private link endpoint which is 10.0.2.5 in this example.
+4. A private connection to the storage account is established through the private link endpoint.
+
+## Getting Started
+
+There are a couple of prerequisites you will need:
+
+- A service principal. You will need the service principal app id, object id and password
+- A SSH key pair
+
+### Creating a service principal
+
+You must pass the object id of a service principal as a parameter. This SP will be given Storage Blob Data Contributor Role Assignment at the Resource Group level. This SP can be used to connect to the storage accounts.
+
+To create a service principal, follow the guidance on [creating a password-based authenticated sp here](https://docs.microsoft.com/cli/azure/create-an-azure-service-principal-azure-cli#password-based-authentication). The following is an example:
+
+```bash
+az ad sp create-for-rbac --scopes /subscriptions/mySubscriptionID \
+  --name myServicePrincipalName \
+  --role rolename
+
+az ad sp create-for-rbac --scopes "/subscriptions/mySubscriptionID" --role "Contributor" --name mySpName
+{
+  "appId": "appIdGuid",
+  "displayName": "mySpName",
+  "name": "guid",
+  "password": "password",
+  "tenant": "myTenantId"
+}
+```

```

Comment:
Mentioned below, but it would be nice to do this via role assignments to the user doing the deployment instead -- just to keep it simple, and to not pollute the customer's AAD instance with "yet another SP"